elements/lisk-tree/src/sparse_merkle_tree/sparse_merkle_tree.ts

```diff
@@ -35,8 +43,108 @@ export class SparseMerkleTree {
 	public get db(): Database {
 		return this._db;
 	}
+
+	public async getNode(nodeHash: Buffer): Promise<TreeNode> {
+		const data = await this._hashToValueMap.get(nodeHash);
+
+		if (!data) {
+			throw new Error(`Hash does not exist in merkle tree: ${nodeHash.toString('hex')}`);
+		}
+
+		if (isLeaf(data)) {
+			const { key, value } = getLeafData(data, this.keyLength);
+			return new Leaf(key, value);
+		}
+
+		const { leftHash, rightHash } = getBranchData(data);
+
+		return new Branch(leftHash, rightHash);
+	}
+
+	public async update(key: Buffer, value: Buffer) {
+		if (!value) {
+			throw new Error('Value cannot be empty');
+		}
+
+		if (key.length !== this.keyLength) {
+			throw new Error(`Key is not equal to defined key length of ${this.keyLength}`);
+		}
+
+		const newLeaf = new Leaf(key, value);
+		const ancestorNodes: TreeNode[] = [];
+		let bottomNode: TreeNode = new Empty();
+		let currentNode = await this.getNode(this.rootHash);
+		const binaryKey = binaryExpansion(key, this.keyLength);
+		let h = 0;
+
+		while (!isLeaf(currentNode.hash)) {
```

Comment:
This function `isLeaf` expect `value` not the `hash`. From hash we can't check the type of the node as the prefix was concatenated before hashing.

Reply:
I did changed it but somehow I didn't push the changes 🤔